toolkit/scripts/check_required_kernel_configs.py

```diff
@@ -0,0 +1,81 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+# Script to check for required kernel configs in a kernel config file
+# Usage: python3 check_required_kernel_configs.py --required_configs <path to json of required configs> --config_file <path to config being checked>
+
+import json
+import argparse
+import sys
+
+def check_config_arch(input_file):
+    with open(input_file, 'r') as file:
+        contents = file.read()
+    file.close()
+    if "Linux/x86_64" in contents:
+        return "AMD64"
+    elif "Linux/arm64" in contents:
+        return "ARM64"
+    else:
+        return None
+
+def check_strings_in_file(json_file, input_file):
+    arch = check_config_arch(input_file)
+    with open(json_file, 'r') as file:
+        data = json.load(file)
+        configData = data['required-configs']
+
+    with open(input_file, 'r') as file:
+        contents = file.read().split("\n")
+
+    missing_configs = {}
+
+    for key, value in configData.items():
+        # check for arch
+        if arch not in value['arch']:
+            continue
+        found = False
+        for line in contents:
+            # check for config in line (without extra _VALUE)
+            if "{0}=".format(key) in line or "{0} is not set".format(key) in line:
+                for val in value['value']:
+                    if val in line and val != "":
+                        found = True
+                        break
+                else:
+                    missing_configs[key] = (line.split(key)[1].replace('=',''), value['value'], value['comment'], value['PR'])
```

Comment:
>missing_configs [](http://example.com/codeflow?start=20&length=15)

Could we use an object with well-named fields instead? That way it's going to be easier to understand what's going on instead of reading raw indeces.

Example:
```python
missing_configs[key] = {"config_name" = key, "allowed_values" = value.value, "comment" = value.comment, "pr" = value.PR}
```

Reply:
Erm, what I had in mind, is just making the field names visible, so the reader understands what they are for with more ease.:) I think my comment was too vague - I apologize.